src/lib/query.ts

```diff
@@ -395,22 +400,44 @@ public static async bulkQueryProductIds (conn: Connection, productList: Set<Stri
-            }
-            let paramsObject2: Query={     
-                "queryBegining": "SELECT Pricebook2.enxCPQ__TECH_External_Id__c, Product2.enxCPQ__TECH_External_Id__c, CurrencyIsoCode, Pricebook2Id, Product2Id," + this.pbeQuery + " FROM PricebookEntry WHERE Product2.enxCPQ__Root_Product__r.Name IN (",
-                "queryConditions": ") AND Pricebook2.IsStandard = true AND Product2.RecordType.Name != 'Charge Element'",
-                "objectsList": productList,
-                "sobjectName": "standard PricebookEntry"
+            let paramsObject1: Query;
+            let paramsObject2: Query;
+            if(!this.currenciesList){
+                  paramsObject1 ={     
+                    "queryBegining": "SELECT Pricebook2.enxCPQ__TECH_External_Id__c, Product2.enxCPQ__TECH_External_Id__c, CurrencyIsoCode, Pricebook2Id, Product2Id," + this.pbeQuery + " FROM PricebookEntry WHERE Product2.Name IN (",
+                    "queryConditions": ") AND Pricebook2.IsStandard = true AND Product2.RecordType.Name != 'Charge Element'",
+                    "objectsList": productList,
+                    "sobjectName": "standard PricebookEntry",
+                }
+                paramsObject2 ={     
+                    "queryBegining": "SELECT Pricebook2.enxCPQ__TECH_External_Id__c, Product2.enxCPQ__TECH_External_Id__c, CurrencyIsoCode, Pricebook2Id, Product2Id," + this.pbeQuery + " FROM PricebookEntry WHERE Product2.enxCPQ__Root_Product__r.Name IN (",
+                    "queryConditions": ") AND Pricebook2.IsStandard = true AND Product2.RecordType.Name != 'Charge Element'",
+                    "objectsList": productList,
+                    "sobjectName": "standard PricebookEntry"
+                }
+            }else{
+                paramsObject1 ={     
+                    "queryBegining": "SELECT Pricebook2.enxCPQ__TECH_External_Id__c, Product2.enxCPQ__TECH_External_Id__c, CurrencyIsoCode, Pricebook2Id, Product2Id," + this.pbeQuery + " FROM PricebookEntry WHERE Product2.Name IN (",
+                    "queryConditions": ") AND Pricebook2.IsStandard = true AND Product2.RecordType.Name != 'Charge Element' AND CurrencyIsoCode IN (" + Util.setToIdString(this.currenciesList) + ") ",
+                    "objectsList": productList,
+                    "sobjectName": "standard PricebookEntry"
+                }
+                paramsObject2 ={     
+                    "queryBegining": "SELECT Pricebook2.enxCPQ__TECH_External_Id__c, Product2.enxCPQ__TECH_External_Id__c, CurrencyIsoCode, Pricebook2Id, Product2Id," + this.pbeQuery + " FROM PricebookEntry WHERE Product2.enxCPQ__Root_Product__r.Name IN (",
+                    "queryConditions": ") AND Pricebook2.IsStandard = true AND Product2.RecordType.Name != 'Charge Element' AND CurrencyIsoCode IN (" + Util.setToIdString(this.currenciesList) + ") ",
+                    "objectsList": productList,
+                    "sobjectName": "standard PricebookEntry"
+                }
```

Comment:
Maybe it's not so important, but I don't like the way how this is done. I'd separate that somehow. Maybe consts or separate methods. I'd look for some way of hiding that stuff.